In [1]:
# YOLOv8 Training Script

from ultralytics import YOLO
import os
from datetime import datetime

# Load a pretrained YOLOv8n variant (custom lightweight model)
model = YOLO('../src/human_detector/models/yolov11n.pt')

# Training configuration
epochs = 50
imgsz = 416
batch_size = 2

# Dynamically name experiment folder
time_stamp = datetime.now().strftime("%Y%m%d_%H%M")
exp_name = f"yolov11n_ep{epochs}_imgsz{imgsz}_bs{batch_size}_{time_stamp}"

# Train the model with inline arguments
results = model.train(
    data='data.yaml',
    epochs=epochs,
    patience=20,
    imgsz=imgsz,
    batch=batch_size,
    device=0,
    half=True,
    workers=2,
    cache=False,

    # Low-light augmentation settings
    hsv_h=0.015,
    hsv_s=0.7,
    hsv_v=0.4,
    translate=0.1,
    scale=0.5,
    fliplr=0.5,
    mosaic=1.0,
    mixup=0.0,
    copy_paste=0.0,

    # Output settings
    project='results',
    name=exp_name
)

# Save trained model path
save_dir = str(results.save_dir)
best_pt = os.path.join(save_dir, 'weights', 'best.pt')
print(f"✅  Trained checkpoint saved to: {best_pt}")


New https://pypi.org/project/ultralytics/8.3.143 available 😃 Update with 'pip install -U ultralytics'
Ultralytics 8.3.141 🚀 Python-3.10.12 torch-2.7.0+cu126 CUDA:0 (NVIDIA GeForce MX550, 1864MiB)
engine/trainer: agnostic_nms=False, amp=True, augment=False, auto_augment=randaugment, batch=2, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=data.yaml, degrees=0.0, deterministic=True, device=0, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, epochs=50, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=True, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=416, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=../src/human_detector/models/yolov11n.pt, momentum=0.937, mosaic=1.0, multi_scale=False, name=yolov11n_ep50_imgsz416_bs2_2025052

train: Scanning /home/yogee/Desktop/human_detector_ws/training_data/data_person/train/labels.cache... 2793 images, 23 backgrounds, 0 corrupt: 100%|██████████| 2793/2793 [00:00<?, ?it/s]


val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 2846.8±2391.7 MB/s, size: 252.0 KB)


val: Scanning /home/yogee/Desktop/human_detector_ws/training_data/data_person/test/labels.cache... 1407 images, 44 backgrounds, 0 corrupt: 100%|██████████| 1407/1407 [00:00<?, ?it/s]


Plotting labels to results/yolov11n_ep50_imgsz416_bs2_20250523_1040/labels.jpg... 


/home/yogee/.local/lib/python3.10/site-packages/matplotlib/projections/__init__.py:63: UserWarning: Unable to import Axes3D. This may be due to multiple versions of Matplotlib being installed (e.g. as a system package and as a pip package). As a result, the 3D projection is not available.
  warnings.warn("Unable to import Axes3D. This may be due to multiple versions of "


optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.001667, momentum=0.9) with parameter groups 81 weight(decay=0.0), 88 weight(decay=0.0005), 87 bias(decay=0.0)
Image sizes 416 train, 416 val
Using 2 dataloader workers
Logging results to results/yolov11n_ep50_imgsz416_bs2_20250523_1040
Starting training for 50 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       1/50     0.232G       1.22       1.64      1.207          4        416: 100%|██████████| 1397/1397 [02:17<00:00, 10.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:19<00:00, 18.17it/s]


                   all       1407       3358      0.801      0.732      0.733      0.539

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       2/50      0.26G      1.249      1.351      1.252          1        416: 100%|██████████| 1397/1397 [02:26<00:00,  9.50it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:32<00:00, 10.88it/s]


                   all       1407       3358      0.806      0.742      0.742      0.561

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       3/50     0.293G      1.252      1.234      1.263         11        416: 100%|██████████| 1397/1397 [02:31<00:00,  9.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:20<00:00, 17.59it/s]


                   all       1407       3358      0.853      0.733      0.768      0.572

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       4/50     0.305G      1.222      1.158      1.247          1        416: 100%|██████████| 1397/1397 [02:21<00:00,  9.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:20<00:00, 17.48it/s]


                   all       1407       3358      0.843      0.739      0.775      0.576

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       5/50     0.318G      1.187       1.09      1.243          4        416: 100%|██████████| 1397/1397 [02:19<00:00, 10.00it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:19<00:00, 17.67it/s]


                   all       1407       3358      0.774      0.732      0.734      0.546

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       6/50     0.318G      1.134       1.03      1.201          7        416: 100%|██████████| 1397/1397 [02:21<00:00,  9.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:20<00:00, 16.85it/s]

                   all       1407       3358      0.842      0.749      0.772      0.581



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       7/50     0.318G      1.129     0.9977      1.206          7        416: 100%|██████████| 1397/1397 [02:17<00:00, 10.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:19<00:00, 17.68it/s]


                   all       1407       3358      0.877      0.752      0.789        0.6

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       8/50     0.318G       1.08      0.965      1.182          1        416: 100%|██████████| 1397/1397 [02:21<00:00,  9.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:20<00:00, 17.50it/s]

                   all       1407       3358      0.845      0.741      0.778      0.612



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       9/50     0.318G      1.072     0.9189      1.175          1        416: 100%|██████████| 1397/1397 [02:20<00:00,  9.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:19<00:00, 18.13it/s]


                   all       1407       3358      0.866      0.759      0.792       0.61

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      10/50     0.318G       1.05     0.9068       1.17          7        416: 100%|██████████| 1397/1397 [02:21<00:00,  9.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:19<00:00, 17.77it/s]


                   all       1407       3358      0.879      0.766      0.798      0.637

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      11/50     0.318G      1.033     0.8858      1.156         16        416: 100%|██████████| 1397/1397 [02:18<00:00, 10.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:20<00:00, 17.27it/s]


                   all       1407       3358      0.869      0.772      0.801      0.635

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      12/50     0.318G      1.024     0.8571      1.141          1        416: 100%|██████████| 1397/1397 [02:20<00:00,  9.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:20<00:00, 17.33it/s]

                   all       1407       3358      0.871      0.773      0.803      0.637



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      13/50      0.33G      1.001     0.8367      1.139          5        416: 100%|██████████| 1397/1397 [02:20<00:00,  9.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:19<00:00, 17.90it/s]


                   all       1407       3358      0.883       0.77      0.806      0.641

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      14/50      0.33G     0.9774     0.8121       1.13          7        416: 100%|██████████| 1397/1397 [02:19<00:00,  9.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:19<00:00, 18.02it/s]

                   all       1407       3358      0.873      0.776      0.808      0.648



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      15/50      0.33G      0.984     0.8174      1.134          7        416: 100%|██████████| 1397/1397 [02:19<00:00, 10.04it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:20<00:00, 17.58it/s]

                   all       1407       3358      0.864      0.775       0.81      0.647



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      16/50      0.33G     0.9748     0.8025      1.124         29        416: 100%|██████████| 1397/1397 [02:18<00:00, 10.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:19<00:00, 17.75it/s]


                   all       1407       3358      0.875      0.775      0.807      0.645

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      17/50      0.33G     0.9556     0.7968      1.121          7        416: 100%|██████████| 1397/1397 [02:19<00:00, 10.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:19<00:00, 17.80it/s]

                   all       1407       3358      0.882      0.777      0.811      0.658



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      18/50      0.33G     0.9388     0.7913      1.106          5        416: 100%|██████████| 1397/1397 [02:20<00:00,  9.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:20<00:00, 17.57it/s]

                   all       1407       3358      0.884      0.777       0.81      0.652



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      19/50      0.33G     0.9336     0.7469      1.106          9        416: 100%|██████████| 1397/1397 [02:19<00:00, 10.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.83it/s]

                   all       1407       3358      0.886      0.779      0.815      0.654



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      20/50      0.33G     0.9339     0.7464      1.099          0        416: 100%|██████████| 1397/1397 [02:14<00:00, 10.39it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.87it/s]

                   all       1407       3358      0.883      0.783      0.815      0.661



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      21/50      0.33G     0.9216     0.7417      1.098         10        416: 100%|██████████| 1397/1397 [02:14<00:00, 10.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:19<00:00, 18.51it/s]


                   all       1407       3358      0.889      0.782      0.816      0.659

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      22/50      0.33G     0.9116     0.7333      1.099          1        416: 100%|██████████| 1397/1397 [02:17<00:00, 10.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.86it/s]

                   all       1407       3358      0.874      0.787      0.816      0.665



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      23/50      0.33G     0.9036     0.7154      1.093          5        416: 100%|██████████| 1397/1397 [02:14<00:00, 10.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:19<00:00, 18.45it/s]

                   all       1407       3358      0.896      0.774      0.818      0.672



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      24/50      0.33G     0.9074     0.7143      1.094         11        416: 100%|██████████| 1397/1397 [02:16<00:00, 10.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.85it/s]

                   all       1407       3358      0.898      0.785      0.824      0.673



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      25/50      0.33G     0.9003     0.7106      1.089          5        416: 100%|██████████| 1397/1397 [02:16<00:00, 10.23it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.65it/s]

                   all       1407       3358      0.904       0.78      0.826       0.67



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      26/50      0.33G     0.8843      0.691      1.075          3        416: 100%|██████████| 1397/1397 [02:16<00:00, 10.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.72it/s]

                   all       1407       3358      0.888      0.786      0.823      0.675



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      27/50      0.33G     0.8756     0.6819      1.078          2        416: 100%|██████████| 1397/1397 [02:15<00:00, 10.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.87it/s]

                   all       1407       3358      0.897       0.79      0.823      0.672



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      28/50      0.33G     0.8587     0.6701      1.064          6        416: 100%|██████████| 1397/1397 [02:15<00:00, 10.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.64it/s]

                   all       1407       3358        0.9      0.792      0.829      0.683



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      29/50      0.33G     0.8565     0.6846      1.067          4        416: 100%|██████████| 1397/1397 [02:14<00:00, 10.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.94it/s]

                   all       1407       3358      0.901      0.791      0.829      0.687



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      30/50      0.33G     0.8544     0.6703      1.064          3        416: 100%|██████████| 1397/1397 [02:15<00:00, 10.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.89it/s]

                   all       1407       3358      0.899       0.79      0.827      0.681



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      31/50      0.33G     0.8479     0.6539      1.059          5        416: 100%|██████████| 1397/1397 [02:15<00:00, 10.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.81it/s]

                   all       1407       3358      0.898      0.793      0.828      0.687



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      32/50      0.33G     0.8336     0.6482      1.051         20        416: 100%|██████████| 1397/1397 [02:14<00:00, 10.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.93it/s]

                   all       1407       3358      0.884      0.796      0.829      0.681



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      33/50      0.33G     0.8471     0.6496      1.058          7        416: 100%|██████████| 1397/1397 [02:12<00:00, 10.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.86it/s]


                   all       1407       3358      0.905      0.798      0.832      0.695

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      34/50      0.33G     0.8322      0.638      1.056          5        416: 100%|██████████| 1397/1397 [02:05<00:00, 11.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.87it/s]

                   all       1407       3358       0.89      0.798      0.832      0.694



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      35/50      0.33G     0.8131     0.6292      1.047          8        416: 100%|██████████| 1397/1397 [02:06<00:00, 11.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.81it/s]


                   all       1407       3358      0.897      0.795      0.832      0.696

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      36/50      0.33G     0.8241     0.6451      1.049          2        416: 100%|██████████| 1397/1397 [02:05<00:00, 11.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 19.01it/s]

                   all       1407       3358      0.909      0.791       0.83      0.691



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      37/50      0.33G     0.7989     0.6105      1.037          5        416: 100%|██████████| 1397/1397 [02:05<00:00, 11.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 19.00it/s]

                   all       1407       3358      0.903      0.794      0.832      0.692



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      38/50      0.33G     0.8036     0.6186      1.043          5        416: 100%|██████████| 1397/1397 [02:05<00:00, 11.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.64it/s]

                   all       1407       3358      0.895      0.801      0.832      0.697



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      39/50      0.33G     0.7899     0.5988      1.037          5        416: 100%|██████████| 1397/1397 [02:05<00:00, 11.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.80it/s]

                   all       1407       3358      0.919      0.789      0.832      0.695



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      40/50      0.33G     0.8004     0.6083      1.042          3        416: 100%|██████████| 1397/1397 [02:05<00:00, 11.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.60it/s]

                   all       1407       3358      0.903      0.794      0.832      0.697


Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      41/50      0.33G     0.7607     0.5449     0.9955          1        416: 100%|██████████| 1397/1397 [02:05<00:00, 11.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.74it/s]

                   all       1407       3358       0.91      0.788       0.83      0.698



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      42/50      0.33G     0.7455     0.5253     0.9936          4        416: 100%|██████████| 1397/1397 [02:05<00:00, 11.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.93it/s]

                   all       1407       3358       0.91      0.791      0.832      0.699



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      43/50      0.33G     0.7376     0.5141     0.9881          1        416: 100%|██████████| 1397/1397 [02:05<00:00, 11.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.65it/s]

                   all       1407       3358      0.903      0.797      0.832      0.704



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      44/50      0.33G     0.7296     0.5046     0.9874          2        416: 100%|██████████| 1397/1397 [02:06<00:00, 11.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.89it/s]

                   all       1407       3358      0.914      0.794      0.836      0.707



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      45/50      0.33G     0.7203     0.5046     0.9816          1        416: 100%|██████████| 1397/1397 [02:05<00:00, 11.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.68it/s]

                   all       1407       3358      0.912       0.79      0.834      0.703



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      46/50      0.33G     0.7133     0.4894     0.9832          1        416: 100%|██████████| 1397/1397 [02:05<00:00, 11.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 19.06it/s]

                   all       1407       3358      0.909      0.796      0.834      0.705



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      47/50      0.33G     0.7006     0.4854     0.9731          1        416: 100%|██████████| 1397/1397 [02:04<00:00, 11.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 19.05it/s]

                   all       1407       3358      0.906      0.798      0.835      0.707



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      48/50      0.33G     0.7019     0.4781     0.9658          4        416: 100%|██████████| 1397/1397 [02:04<00:00, 11.22it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 18.90it/s]

                   all       1407       3358      0.914      0.794      0.834      0.707



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      49/50      0.33G     0.6905     0.4789     0.9654          1        416: 100%|██████████| 1397/1397 [02:04<00:00, 11.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 19.12it/s]

                   all       1407       3358      0.912      0.797      0.837      0.711



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      50/50      0.33G     0.6886     0.4727     0.9615          1        416: 100%|██████████| 1397/1397 [02:04<00:00, 11.23it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:18<00:00, 19.00it/s]

                   all       1407       3358      0.914      0.796      0.837      0.711



50 epochs completed in 2.135 hours.
Optimizer stripped from results/yolov11n_ep50_imgsz416_bs2_20250523_1040/weights/last.pt, 5.4MB
Optimizer stripped from results/yolov11n_ep50_imgsz416_bs2_20250523_1040/weights/best.pt, 5.4MB

Validating results/yolov11n_ep50_imgsz416_bs2_20250523_1040/weights/best.pt...
Ultralytics 8.3.141 🚀 Python-3.10.12 torch-2.7.0+cu126 CUDA:0 (NVIDIA GeForce MX550, 1864MiB)
YOLO11n summary (fused): 100 layers, 2,582,542 parameters, 0 gradients, 6.3 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 352/352 [00:16<00:00, 21.09it/s]


                   all       1407       3358      0.912      0.797      0.837      0.711
                person       1250       2734      0.841      0.597      0.679       0.48
                teleco        331        624      0.983      0.997      0.995      0.941
Speed: 0.2ms preprocess, 7.2ms inference, 0.0ms loss, 0.9ms postprocess per image
Results saved to results/yolov11n_ep50_imgsz416_bs2_20250523_1040
✅  Trained checkpoint saved to: results/yolov11n_ep50_imgsz416_bs2_20250523_1040/weights/best.pt


# 🧠 YOLOv8 Training Metrics Cheatsheet

## 🔢 Loss Curves

| Metric            | Meaning                                                | Goal / Good Trend                |
|-------------------|--------------------------------------------------------|----------------------------------|
| `train/box_loss`  | Bounding box regression loss (training)                | ↓ Steady decline                 |
| `train/cls_loss`  | Classification loss (training)                         | ↓ Should fall and stabilize     |
| `train/dfl_loss`  | Distribution Focal Loss (for bounding box quality)     | ↓ Smooth drop                   |
| `val/box_loss`    | Box loss on validation set                             | ↓ Best indicator of generalization |
| `val/cls_loss`    | Classification loss on validation                      | ↓ Falling → Good model fit      |
| `val/dfl_loss`    | DFL loss on validation                                 | ↓ Matches train trend = no overfit |

---

## 📈 Detection Metrics

| Metric                | Description                                                 | Goal / Interpretation           |
|------------------------|-------------------------------------------------------------|----------------------------------|
| `metrics/precision(B)` | % of correct detections out of all predictions              | ↑ High = few false positives     |
| `metrics/recall(B)`    | % of actual objects detected                                | ↑ High = few missed detections   |
| `metrics/mAP50(B)`     | Accuracy of boxes at 50% IoU threshold                     | ↑ Main object detection metric   |
| `metrics/mAP50-95(B)`  | Average mAP over IoU 0.5 to 0.95 (stricter)                | ↑ Robust performance measure     |

---

## 🧪 Ideal Curve Patterns

- **Losses**: should steadily decrease.
- **Validation loss** rising while training loss decreases → possible **overfitting**.
- **Precision & Recall**: should increase and plateau near 1.0.
- **mAP**: aim for **>0.90 mAP50** and **>0.80 mAP50-95** for high-quality results.

---

## ⚠️ Tips

- 🧊 Use `half=True` for mixed precision training on supported GPUs.
- 🧠 Watch `val/box_loss` and `mAP50-95` to catch overfitting early.
- 🧪 If metrics plateau early, try increasing `epochs`, `imgsz`, or improving labels.
- 💾 Check `runs/train/.../results.png` for curve visualization.

---
